torch/_refs/__init__.py

```diff
@@ -520,6 +520,50 @@ def log10(a):
     return prims.log10(a)
 
 
+@out_wrapper
+def log_softmax(
+    a: TensorLikeType,
+    dim: int,
+    *,
+    dtype: Optional[torch.dtype] = None,
+) -> TensorLikeType:
+    result_dtype = dtype or a.dtype
+    computation_dtype = utils.get_computation_dtype(a.dtype)
+    a = prims.convert_element_type(a, computation_dtype)
+    a_max = amax(a, dim, keepdim=True)
+    shifted = a - a_max
```

Comment:
Nit: comment for why the shift occurs would be nice

Reply:
Oh, the shift is actually not required here because stabilized logsumexp is used. I will remove it.